### ISSUE TITLE
Refactor includeForks and optimizeForDisabled into workers layer

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#7162]
Currently the logic for checking for `includeForks=true` or `enabled=false` is duplicated in every platform's `initRepo()` code. After #7161, the logic should be moved to our workers/repo/init logic and done immediately after `platform.initRepo()` is called.

**Comment by zharinov** [comment]
Superseded by #7364
